raster/src/main/scala/geotrellis/raster/histogram/StreamingHistogram.scala

```diff
@@ -462,64 +478,81 @@ class StreamingHistogram(
+      val pdf = bs.map(_.count.toDouble / n)
+      val cdf = pdf.scanLeft(0.0)(_ + _).drop(1)
+      val data = ds.zip(cdf).sliding(2)
+
+      data.map({ ab => (ab.head, ab.tail.head) })
+    }
   }
 
   /**
     * Get the (approximate) percentile of this item.
     */
-  def percentileRanking(item: Double): Double = {
-    val data = cdfIntervals
-    val tt = data.dropWhile(_._2._1 <= item).next
-    val (d1, pct1) = tt._1
-    val (d2, pct2) = tt._2
-    val x = (item - d1) / (d2 - d1)
-
-    (1-x)*pct1 + x*pct2
-  }
+  def percentileRanking(item: Double): Double =
+    if(buckets.size == 1) {
+      if(item < buckets.head.label) 0.0 else 1.0
+    } else {
+      val data = cdfIntervals
+      val tt = data.dropWhile(_._2._1 <= item).next
+      val (d1, pct1) = tt._1
+      val (d2, pct2) = tt._2
+      if(item - d1 < 0.0) {
+        0.0
+      } else {
+        val x = (item - d1) / (d2 - d1)
+
+        (1-x)*pct1 + x*pct2
+      }
+    }
 
   /**
     * For each q in qs, all between 0 and 1, find a number
     * (approximately) at the qth percentile.
     */
-  def percentileBreaks(qs: Seq[Double]): Seq[Double] = {
-    val data = cdfIntervals
-    if(!data.hasNext) {
-      Seq()
+  def percentileBreaks(qs: Seq[Double]): Seq[Double] =
+    if(buckets.size == 1) {
+      qs.map(z => buckets.head.label)
     } else {
-      val result = MutableListBuffer[Double]()
-      var curr = data.next
-      def getValue(q: Double): Double = {
-        val (d1, pct1) = curr._1
-        val (d2, pct2) = curr._2
-        val x = (q - pct1) / (pct2 - pct1)
-
-        (1-x)*d1 + x*d2
-      }
-
-      for(q <- qs) {
-        if (q == 0.0) { result += minValue().getOrElse(Double.NegativeInfinity) }
-        else if (q == 1.0) { result += maxValue().getOrElse(Double.PositiveInfinity) }
-        else {
-          if(q < curr._2._2) {
-            result += getValue(q)
+      val data = cdfIntervals
+      if(!data.hasNext) {
+        Seq()
+      } else {
+        val result = MutableListBuffer[Double]()
+        var curr = data.next
+        def getValue(q: Double): Double = {
+          val (d1, pct1) = curr._1
+          val (d2, pct2) = curr._2
+          if(q - pct1 < 0.0) {
+            minValue().getOrElse(Double.NegativeInfinity)
           } else {
-            while(data.hasNext && curr._2._2 <= q) { curr = data.next }
-            result += getValue(q)
+            val x = (q - pct1) / (pct2 - pct1)
+            (1 - x) * d1 + x * d2
           }
         }
-      }
 
-      result.toSeq
+        for(q <- qs) {
+          if (q == 0.0) { result += minValue().getOrElse(Double.NegativeInfinity) }
+          else if (q == 1.0) { result += maxValue().getOrElse(Double.PositiveInfinity) }
+          else {
+            if(q < curr._2._2) {
+              result += getValue(q)
+            } else {
+              while(data.hasNext && curr._2._2 <= q) { curr = data.next }
+              result += getValue(q)
+            }
+          }
+        }
```

Comment:
The algorithm has changed to use the current val until the quantile passes the current value's quantile.